app/controllers/posts_controller.rb

```diff
@@ -20,7 +20,10 @@ def create
   private
 
   def timeline_posts
-    @timeline_posts ||= Post.all.ordered_by_most_recent.includes(:user)
+    users_posts = current_user.posts
+    friends_posts = current_user.friends.map { |friend| friend.posts if friend.posts.any? }.compact
+    friends_posts = friends_posts.flatten
+    @timeline_posts = users_posts + friends_posts
```

Comment:
this should be handled at model level.
Why:
It would be simpler would allow for improved performance and avoid the use of Enumerables ( as seen in line 23)

Something in User model likefollowing would do:

```ruby
def timeline_posts
  Post.where(user: (self.friends + self))
end